RestaurantReservation.API/Controllers/CustomersController.cs

```diff
@@ -0,0 +1,144 @@
+﻿using AuthenticationService;
+using AutoMapper;
+using Microsoft.AspNetCore.Authorization;
+using Microsoft.AspNetCore.Mvc;
+using RestaurantReservation.API.Models.Customer;
+using RestaurantReservation.API.Utilities;
+using RestaurantReservation.Db.Entities;
+using RestaurantReservation.Db.Interfaces;
+
+namespace RestaurantReservation.Api.Controllers;
+
+[Authorize]
+[Route("api/customers")]
+[ApiController]
+public class CustomersController : ControllerBase
+{
+    private readonly ICustomerRepository _customerRepository;
+    private readonly IMapper _mapper;
+
+    public CustomersController(ICustomerRepository customerRepository, IMapper mapper)
+    {
+        _customerRepository = customerRepository;
+        _mapper = mapper;
+    }
+
+    /// <summary>
+    /// Gets all customers.
+    /// </summary>
+    /// <returns>A list of customer data transfer objects.</returns>
+    /// <response code="200">Returns the list of customers.</response>
+    [HttpGet]
+    [ProducesResponseType(StatusCodes.Status200OK, Type = typeof(IEnumerable<CustomerDto>))]
+    public async Task<ActionResult<IEnumerable<CustomerDto>>> GetCustomers()
+    {
+        var customers = await _customerRepository.GetAllCustomersAsync();
+        return Ok(_mapper.Map<IEnumerable<CustomerDto>>(customers));
```

Comment:
we should introduce a service layer to handle any additional logic and manage the mapping. It is not best practice for the controller to directly depend on repositories